File: staGledas.Model/Models/Pretplate.cs
using System;

namespace staGledas.Model.Models
{
    public class Pretplate
    {
        public int Id { get; set; }
        public int KorisnikId { get; set; }
        public string? StripeCustomerId { get; set; }
        public string? StripeSubscriptionId { get; set; }
        public string? Status { get; set; }
        public DateTime? DatumPocetka { get; set; }
        public DateTime? DatumIsteka { get; set; }
        public DateTime DatumKreiranja { get; set; }
        public virtual Korisnici? Korisnik { get; set; }
    }
}

File: staGledas-UI/stagledas_mobile/lib/models/pretplate.dart
import 'package:json_annotation/json_annotation.dart';

part 'pretplate.g.dart';

@JsonSerializable()
class Pretplate {
  int? id;
  int? korisnikId;
  String? stripeCustomerId;
  String? stripeSubscriptionId;
  String? status;
  DateTime? datumPocetka;
  DateTime? datumIsteka;
  DateTime? datumKreiranja;
  DateTime? datumIzmjene;

  Pretplate({
    this.id,
    this.korisnikId,
    this.stripeCustomerId,
    this.stripeSubscriptionId,
    this.status,
    this.datumPocetka,
    this.datumIsteka,
    this.datumKreiranja,
    this.datumIzmjene,
  });

  factory Pretplate.fromJson(Map<String, dynamic> json) =>
      _$PretplateFromJson(json);

  Map<String, dynamic> toJson() => _$PretplateToJson(this);

  bool get isActive => status == 'active';
}
